### PR TITLE
Adding YOLO examples to ART

### PR DESCRIPTION
**Pull request by kieranfraser**
Signed-off-by: Kieran Fraser <kieran.fraser@ibm.com>

# Description

Demonstrating YOLO v3 and YOLO v5 support within ART. Addition of "get started" files for each which demonstrates how to instantiate YOLO object detectors with ART, run a PGD attack and visually evaluate the result. Addition of example notebooks demonstrating patch attacks on YOLO v3 and YOLO v5.

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: examples/get_started_yolo.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+plt.show()
+
+threshold = 0.85 # 0.5
+dets = detector.predict(x)
+preds = extract_predictions(dets[0], threshold)
+plot_image_with_boxes(img=img, boxes=preds[1], pred_cls=preds[0], title="Predictions on original image")
+
+dets = detector.predict(image_adv)
+preds = extract_predictions(dets[0], threshold)
+plot_image_with_boxes(img=image_adv[0].transpose(1,2,0).copy(), boxes=preds[1], pred_cls=preds[0], title="Predictions on adversarial image")
```
Comment (beat-buesser) [inline examples/get_started_yolo.py:209]: Please add empty line at the end of the file.

File: examples/get_started_yolo.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+'''
+MODEL = 'yolov3' # OR yolov5
+
+
+if MODEL == 'yolov3':
+
+    from pytorchyolo.utils.loss import compute_loss
+    from pytorchyolo.models import load_model
+
+    class Yolo(torch.nn.Module):
```
Comment (beat-buesser) [inline examples/get_started_yolo.py:111]: Could you please update the docstrings of `PyTorchYolo` where the model is input to mention that a wrapper is required as shown in this example?

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi Kieran, thank you very much for your first pull request to ART!

I have left a few comments above, what do you think?

Could you also please check the reports of the style checks?

Could you please run the notebook for more iterations, e.g. 1000 with optimizer `pgd` and learning rate of 1?

**Comment by kieranfraser** [comment]
> Hi @kieranfraser , thank you very much for your first pull request to ART!
> 
> I have left a few comments above, what do you think?
> 
> Could you also please check the reports of the style checks?
> 
> Could you please run the notebook for more iterations, e.g. 1000 with optimizer `pgd` and learning rate of 1?

Thanks @beat-buesser! 

I have included your suggestions in the my latest additions and the style checks are now passing. I also ran the notebook with 1000 iterations, optimizer `pgd` and learning rate 1. Should I commit and push that notebook so those results are visible?

**Comment by kieranfraser** [comment]
> > Hi @kieranfraser , thank you very much for your first pull request to ART!
> > I have left a few comments above, what do you think?
> > Could you also please check the reports of the style checks?
> > Could you please run the notebook for more iterations, e.g. 1000 with optimizer `pgd` and learning rate of 1?
> 
> Thanks @beat-buesser!
> 
> I have included your suggestions in my latest additions and the style checks are now passing. I also ran the notebook with 1000 iterations, optimizer `pgd` and learning rate 1. Should I commit and push that notebook so those results are visible?

Added notebook with results of run with 1000 iterations.

**Review by beat-buesser — APPROVED** [review]
Hi @kieranfraser Thank you very much and congratulations for your first accepted pull request to ART!
